app/src/main/java/com/asfoundation/wallet/onboarding/OnboardingViewModel.kt

```diff
@@ -3,33 +3,59 @@ package com.asfoundation.wallet.onboarding
 import com.asfoundation.wallet.base.BaseViewModel
 import com.asfoundation.wallet.base.SideEffect
 import com.asfoundation.wallet.base.ViewState
+import com.asfoundation.wallet.onboarding.use_cases.HasWalletUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
 import javax.inject.Inject
 
 sealed class OnboardingSideEffect : SideEffect {
   object NavigateToRecoverWallet : OnboardingSideEffect()
+  object NavigateToLegalsConsent : OnboardingSideEffect()
+  object NavigateToExit : OnboardingSideEffect()
 }
 
 data class OnboardingState(val pageNumber: Int = 0) : ViewState
 
 @HiltViewModel
-class OnboardingViewModel @Inject constructor() :
+class OnboardingViewModel @Inject constructor(hasWalletUseCase: HasWalletUseCase) :
   BaseViewModel<OnboardingState, OnboardingSideEffect>(initialState()) {
 
   companion object {
     fun initialState(): OnboardingState {
-      return OnboardingState(0)
+      return OnboardingState()
     }
   }
 
-  fun handleBackButtonClick() {
-    setState { copy(pageNumber = 0) }
+  var rootPage = 0
+
+  init {
+    hasWalletUseCase()
+      .observeOn(AndroidSchedulers.mainThread())
```

Comment:
I think we should try to use the RxSchedulers interface for consistency purposes, especially if we want to later start creating tests for this